From 014bef948a65f05fb8fd203edf0fa3859f529f21 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Mon, 1 Jun 2020 00:20:42 +0200
Subject: [PATCH 01/17] ResolverActivity: use hardware rendering for Intent
 Resolver dialog (q-x86)

Workaround to avoid graphic glitches with gbm_gralloc happening
when launcher needs to be selected at first boot

Suggested by Franco Catrin for issues affecting software rendering path:
https://github.com/android-rpi/device_brcm_rpi3/issues/49

(q-x86) Resolved conflict due to changes in q-x86 branch
---
 core/java/com/android/internal/app/ResolverActivity.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/java/com/android/internal/app/ResolverActivity.java b/core/java/com/android/internal/app/ResolverActivity.java
index bef60d97ad9c..d165736bc073 100644
--- a/core/java/com/android/internal/app/ResolverActivity.java
+++ b/core/java/com/android/internal/app/ResolverActivity.java
@@ -69,6 +69,7 @@ import android.view.View;
 import android.view.ViewGroup;
 import android.view.ViewGroup.LayoutParams;
 import android.view.WindowInsets;
+import android.view.WindowManager;
 import android.widget.AbsListView;
 import android.widget.AdapterView;
 import android.widget.BaseAdapter;
@@ -290,6 +291,10 @@ public class ResolverActivity extends Activity {
     protected void onCreate(Bundle savedInstanceState, Intent intent,
             CharSequence title, int defaultTitleRes, Intent[] initialIntents,
             List<ResolveInfo> rList, boolean supportsAlwaysUseOption) {
+
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED,
+            WindowManager.LayoutParams.FLAG_HARDWARE_ACCELERATED);
+
         setTheme(R.style.Theme_DeviceDefault_Resolver);
         super.onCreate(savedInstanceState);
 
-- 
2.17.1

